File: backend/src/main/java/com/smartcampus/backend/dto/weather/AirQualityAtDTO.java
package com.smartcampus.backend.dto.weather;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class AirQualityAtDTO {

    private Double latitude;
    private Double longitude;
    private LocalDate date;
    private LocalTime time;

    // Resume du jour demande (moyennes)
    private Double pm2_5Avg;
    private Double pm10Avg;
    private Double ozoneAvg;
    private Double nitrogenDioxideAvg;

    // Point horaire demande (rempli seulement si "time" est fourni et disponible)
    private Double pm2_5;
    private Double pm10;
    private Double ozone;
    private Double nitrogenDioxide;

    // Indice + traduction (icone / signification)
    private Integer europeanAqi;
    private String category;
    private String description;
    private String icon;
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/weather/AirQualityAtDto.kt
package com.ObservatoireCampus.mobile.model.weather

/**
 * Reponse de /api/air-quality/at.
 * Si "time" n'a pas ete envoye dans la requete : seuls les champs *Avg sont remplis
 * (moyenne du jour). Si "time" a ete envoye : pm2_5/pm10/ozone/nitrogenDioxide et
 * europeanAqi/category/description/icon refletent l'heure precise.
 */
data class AirQualityAtDto(
    val latitude: Double?,
    val longitude: Double?,
    val date: String?,
    val time: String?,
    val pm2_5Avg: Double?,
    val pm10Avg: Double?,
    val ozoneAvg: Double?,
    val nitrogenDioxideAvg: Double?,
    val pm2_5: Double?,
    val pm10: Double?,
    val ozone: Double?,
    val nitrogenDioxide: Double?,
    val europeanAqi: Int?,
    val category: String?,
    val description: String?,
    val icon: String?
)